src/jsifier.mjs

```diff
@@ -440,6 +453,11 @@ function(${args}) {
       compileTimeContext.i53ConversionDeps.forEach((d) => deps.push(d));
     }
 
+    const isAsyncFunction = LibraryManager.library[symbol + '__async'];
+    if (ASYNCIFY && isAsyncFunction == 'auto') {
```

Comment:
For JSPI, I don't think we even need the `handleAsync`. Though there still may be some issues with balanced `runtimeKeepalivePush` `runtimeKeepalivePop`. Maybe worth a try?

Reply:
Yes, I think handleAsync is used to keep the runtime alive using `runtimeKeepalivePush` which I think this currently important.   We can look at further simplification are followups perhaps, but for now I'd rather leave the semantics as matching.